db/inmemory/client.go

```diff
@@ -24,22 +25,19 @@ func (c *client) AddUser(user *domain.User) (string, error){
 	c.ds[user.ID]=user
 	return user.ID,nil
 }
-func (c *client) ListUsers(queryName *string, queryLimit *int32) ([]*domain.User,error){
+func (c *client) ListUsers(queryNameMap map[string]interface{}, queryLimit int32) ([]*domain.User,error){
 	var userInfo =[]*domain.User{}
-	if queryName != nil && queryLimit !=nil{
-		fmt.Println("The queryName and queryLimit are",*queryName,*queryLimit)
-	}
 	var lim int32 = 1
 	for _,value := range c.ds{
-		if queryName !=nil{
-			if *queryName == value.Name{
+		if queryNameMap["name"] !=nil{
+			if strings.ToUpper(fmt.Sprint(queryNameMap["name"])) == strings.ToUpper(value.Name){
```

Comment:
what is the purpose to convert to upper here?

Reply:
Sir if user tries to search a name in any case(upper, lower or mixed) then also he/she should get the filtered object for that name. That's why did like this. Please suggest if this is not supposed to be done or something better can be done.